ibridges/irodsconnector/data_operations.py

```diff
@@ -85,23 +83,35 @@ def get_collection(self, path: IrodsPath) -> irods.collection.iRODSCollection:
             Instance of the collection with `path`.
 
         """
-        if self.collection_exists(path):
-            return self.session.irods_session.collections.get(path)
+        if path.is_collection():
+            return self.session.irods_session.collections.get(str(path))
         raise irods.exception.CollectionDoesNotExist(path)
 
-    def irods_put(self, local_path: str, irods_path: str, resc_name: str = ''):
+    @staticmethod
+    def has_type_dataobj(item) -> bool:
```

Comment:
I think `is_dataobj()` might be a more intuitive name. (idem `has_type_collection()` --> `is_collection()`)

Reply:
I used these function for the IrodsPath verification if a path points to a collections or object. Any suggestions?

Reply:
I now renamed the type functions as you suggested and renamed the IrodsPath functions to dataobject_exists/collection_exists. This should do the trick.